deployment/utils/contract.go

```diff
@@ -0,0 +1,65 @@
+package utils //nolint:revive,nolintlint // TODO: update to meaningful package name
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+	ds "github.com/smartcontractkit/chainlink-deployments-framework/datastore"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	"github.com/xssnick/tonutils-go/address"
+	"github.com/xssnick/tonutils-go/tvm/cell"
+
+	"github.com/smartcontractkit/chainlink-ton/deployment/ccip/operation"
+)
+
+type CompiledContractData struct {
+	Type                           ds.ContractType
+	Code                           *cell.Cell
+	SuggestedTONCoinsForDeployment string
+	ContractVersionSha             string
+	ContractSemver                 *semver.Version
+	ContractPath                   string
+}
+
+type TONContractAddress struct {
+	TONAddress     address.Address
+	CLDFAddressRef ds.AddressRef
+}
+
+// InvokeDeployContractOperation deploys a TON contract if it's not already deployed.
+// It checks the current address, executes the deployment operation if needed,
+// and invokes the provided callback with the new contract address.
+// Returns an error if the deployment fails.
+func InvokeDeployContractOperation(b operations.Bundle, deps operation.TonDeps, chainSelector uint64, currentAddress address.Address, compiledContract CompiledContractData, storage any, messageBody any) (*TONContractAddress, error) {
+	if !currentAddress.IsAddrNone() {
```

Comment:
This check is weird to be in an util method, feels like something a client should check for themselves before calling this fn vs. adding a flag just to return nil if set (not used for anything else).